R/outliers_detect.R

```diff
@@ -10,6 +10,79 @@
 #' outliers_detect(c(1,2,1,2,100))
 outliers_detect <- function(s, method='zscore') {
 
+  if(!is.numeric(s)) {
+    stop("s has to be a numeric vector")
+  }
+
+  if(length(s) < 2) {
+    stop("s must have a length of atleast 2")
+  }
+
+  outliers <- NA
+
+  if (method == "zscore") {
+    outliers <- outliers_detect_zscore(s)
+  } else if (method == "iqr") {
+    outliers <- outliers_detect_iqr(s)
+  } else if (method == "iforest") {
+    outliers <- outliers_detect_iforest(s)
+  } else {
+    stop("Invalid method. Should be ascore, iqr or iforest")
+  }
+
+  outliers
+}
+
+
+#' Detects outliers in a pandas series using isolation forests
+#'
+#' @param s numeric vector for which the outliers need to be found
+#' @param threshold threshold for considering an outlier
+#'
+#' @return boolean vector with indices makerd true for the outliers
+#' @export
+#'
+#' @examples
+#' outliers_detect_iforest(c(1,1,1,1,1,1,1,1,1,1,1e14))
+outliers_detect_iforest <- function(s, threshold = 0.5) {
+  mdl <- isolation.forest(as.data.frame(s), output_score = T)
```

Comment:
Need to namespace here I believe for robustness

```suggestion
  mdl <- isotree::isolation.forest(as.data.frame(s), output_score = T)
```